src/ex_eval.c

```diff
@@ -747,6 +747,43 @@ finish_exception(except_T *excp)
     discard_exception(excp, TRUE);
 }
 
+/*
+ * Save the current exception state in "estate"
+ */
+    void
+exception_state_save(exception_state_T *estate)
+{
+    estate->estate_current_exception = current_exception;
+    estate->estate_did_throw = did_throw;
+    estate->estate_need_rethrow = need_rethrow;
+    estate->estate_trylevel = trylevel;
+}
+
+/*
+ * Restore the current exception state from "estate"
+ */
+    void
+exception_state_restore(exception_state_T *estate)
+{
+    if (current_exception == NULL)
+	current_exception = estate->estate_current_exception;
+    did_throw |= estate->estate_did_throw;
+    need_rethrow |= estate->estate_need_rethrow;
+    trylevel |= estate->estate_trylevel;
```

Comment:
Performing a bitwise operation on `trylevel` looks strange.

Reply:
That is a typo.  I will fix that in the next PR (#13386).